File: ef8-rawsql/AddressBook.Api/Data/Entities/Address.cs
namespace AddressBook.Api.Data.Entities;

public class Address
{
  public int Id { get; set; }
  public string? Name { get; set; }
  public string? Line1 { get; set; }
  public string? Line2 { get; set; }
  public string? Line3 { get; set; }
  public string? CityTown { get; set; }
  public string? StateProvince { get; set; }
  public string? PostalCode { get; set; }
  public string? Country { get; set; }

  public int BookEntryId { get; set; }

}

File: ef8-rawsql/addressbookapp/src/schemas/address.ts
import { object, string, type AnyZodObject } from "zod";

export const addressSchema = object({
  name: string().min(1),
  line1: string().min(1),
  line2: string().optional(),
  line3: string().optional(),
  cityTown: string().min(1),
  stateProvince: string().min(1),
  postalCode: string().min(5),
  country: string().optional()
});

